src/devices/bus/ieee488/hp9122c.cpp

```diff
@@ -0,0 +1,383 @@
+{
+	m_bus->ren_w(this, state);
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_int_w)
+{
+	m_i8291a_irq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::i8291a_dreq_w)
+{
+	m_i8291a_drq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::fdc_intrq_w)
+{
+	m_fdc_irq = state;
+	update_intsel();
+}
+
+WRITE_LINE_MEMBER(hp9122c_device::fdc_drq_w)
+{
+	m_fdc_drq = state;
+	update_intsel();
+}
+
+void hp9122c_device::update_intsel()
+{
+	bool irq = false, firq = false;
+
+	switch (m_intsel & 3) {
+	case 0:
+		/* fdc int connected */
+		irq = m_fdc_drq;
+		firq = m_fdc_irq;
+		break;
+	case 1:
+		/* i8291a ints connected */
+		irq = m_i8291a_drq;
+		firq = m_i8291a_irq;
+		break;
+	case 2:
+		irq = m_index_int;
+		firq = false;
+		break;
+
+	case 3:
+		irq = false;
+		firq = false;
+		break;
+	}
+
+	if (m_cpufirq != firq)
+		m_cpu->set_input_line(M6809_FIRQ_LINE, firq ? ASSERT_LINE : CLEAR_LINE);
+
+	if (m_cpuirq != irq)
+		m_cpu->set_input_line(M6809_IRQ_LINE, irq ? ASSERT_LINE : CLEAR_LINE);
+
+	m_cpuirq = irq;
+	m_cpufirq = firq;
+}
+
+READ8_MEMBER(hp9122c_device::i8291a_dio_r)
+{
+	return m_bus->read_dio();
+}
+
+WRITE8_MEMBER(hp9122c_device::i8291a_dio_w)
+{
+	m_bus->dio_w(this, data);
+}
+
+READ8_MEMBER(hp9122c_device::status_r)
+{
+	uint8_t ret = REG_STATUS_DUAL|REG_STATUS_DISKCHG;
+	auto addr = m_hpib_addr->read();
+
+	if (!m_testmode->read())
+		ret |= 0x10;
+
+	ret |= ((addr & 0x08) << 3) | (addr & 0x07);
+
+	floppy_image_device *floppy = nullptr;
+	if (m_ds0)
+		floppy = m_floppy[0]->get_device();
+	if (m_ds1)
+		floppy = m_floppy[1]->get_device();
+
+	if (floppy) {
+		auto variant = floppy->get_variant();
+		if (variant == floppy_image::DSDD || variant == floppy_image::SSDD)
+			ret |= REG_STATUS_LOW_DENSITY;
+	}
+
+	return ret;
+}
+
+WRITE8_MEMBER(hp9122c_device::cmd_w)
+{
+	floppy_image_device *floppy0 = m_floppy[0]->get_device();
+	floppy_image_device *floppy1 = m_floppy[1]->get_device();
+
+	m_ds0 = !(data & REG_CNTL_DS0) && floppy0;
+	m_ds1 = !(data & REG_CNTL_DS1) && floppy1;
+
+	 if (m_ds0) {
+		 floppy0->mon_w(0);
+		 floppy0->ss_w(!!!(data & REG_CNTL_HEADSEL));
+		 m_fdc->set_floppy(floppy0);
+	 } else if (m_ds1) {
+		 floppy1->mon_w(0);
+		 floppy1->ss_w(!!!(data & REG_CNTL_HEADSEL));
+		 m_fdc->set_floppy(floppy1);
+	}
+
+	if (data & REG_CNTL_CLOCK_SEL)
+		m_fdc->set_clock(XTAL(8'000'000)/4);
+	else
+		m_fdc->set_clock(XTAL(8'000'000)/8);
+
+	m_intsel = ((data & REG_CNTL_INTSEL0) >> 7) | ((data & REG_CNTL_INTSEL1) >> 2);
+	update_intsel();
+}
+
+void hp9122c_device::index_pulse_cb(floppy_image_device *floppy, int state)
+{
+	m_fdc->index_callback(floppy, state);
+
+	if (state)
+		m_index_int = true;
+	update_intsel();
+
+}
+
+WRITE8_MEMBER(hp9122c_device::clridx_w)
+{
+	m_index_int = false;
+	update_intsel();
+}
+
+ROM_START(hp9122c)
+	ROM_REGION(0x10000 , "cpu" , 0)
+	ROM_LOAD("09122-15515.bin" , 0xc000, 0x4000 , CRC(d385e488) SHA1(93b2015037d76cc68b6252df03d6f184104605b9))
```

Comment:
Please use a region the size of the ROM and use `.region` in the memory map to load from the correct offset.